src/main/java/org/springframework/data/mapping/model/SimpleTypeHolder.java

```diff
@@ -32,39 +32,43 @@
  */
 public class SimpleTypeHolder {
 
-	private static final Set<Class<?>> DEFAULTS = new HashSet<>();
-
-	static {
-		DEFAULTS.add(boolean.class);
-		DEFAULTS.add(boolean[].class);
-		DEFAULTS.add(long.class);
-		DEFAULTS.add(long[].class);
-		DEFAULTS.add(short.class);
-		DEFAULTS.add(short[].class);
-		DEFAULTS.add(int.class);
-		DEFAULTS.add(int[].class);
-		DEFAULTS.add(byte.class);
-		DEFAULTS.add(byte[].class);
-		DEFAULTS.add(float.class);
-		DEFAULTS.add(float[].class);
-		DEFAULTS.add(double.class);
-		DEFAULTS.add(double[].class);
-		DEFAULTS.add(char.class);
-		DEFAULTS.add(char[].class);
-		DEFAULTS.add(Boolean.class);
-		DEFAULTS.add(Long.class);
-		DEFAULTS.add(Short.class);
-		DEFAULTS.add(Integer.class);
-		DEFAULTS.add(Byte.class);
-		DEFAULTS.add(Float.class);
-		DEFAULTS.add(Double.class);
-		DEFAULTS.add(Character.class);
-		DEFAULTS.add(String.class);
-		DEFAULTS.add(Date.class);
-		DEFAULTS.add(Locale.class);
-		DEFAULTS.add(Class.class);
-		DEFAULTS.add(Enum.class);
-	}
+	private static final Set<Class<?>> DEFAULTS = new HashSet<Class<?>>() {
```

Comment:
It's the only way you can have a static `DEFAULT` instance see below *and* get the `DEFAULTS` `HashSet` populated already, which is used in the constructor. If `DEFAULTS` is set up in a static initializer block, it's still empty when the constructor is called to create the `DEFAULT` instance.